documents/validator.go

```diff
@@ -465,6 +466,37 @@ func transitionValidator(collaborator identity.DID) Validator {
 	})
 }
 
+// computeFieldsValidator verifies the execution of each compute field by re executing the WASM and checking the result
+// is same as the one that is stored in the document.
+func computeFieldsValidator(timeout time.Duration) Validator {
+	return ValidatorFunc(func(_, new Model) error {
+		computeFields := new.GetComputeFieldsRules()
+		attributes := func() map[AttrKey]Attribute {
+			attrMap := make(map[AttrKey]Attribute)
+			for _, attr := range new.GetAttributes() {
+				attrMap[attr.Key] = attr
+			}
+			return attrMap
+		}()
+
+		for _, computeField := range computeFields {
+			// execute compute fields
+			targetAttr, err := executeComputeField(computeField, attributes, timeout)
+			if err != nil {
+				return err
+			}
+
+			// verify the targetAttr is same as the one already stored
+			attr := attributes[targetAttr.Key]
+			if !reflect.DeepEqual(attr, targetAttr) {
+				return errors.New("compute fields[%s] validation failed", hexutil.Encode(computeField.RuleKey))
```

Comment:
are we keeping errors in the `error.go` file in this directory?

Reply:
We can't do that since this is dynamic error